### PR TITLE
install_klp_product: Fix console switching on s390

### PR DESCRIPTION
**Pull request by mdoucha**
Calling `select_serial_terminal()` more than once doesn't work on s390 workers. Select the best console in `install_klp_product` according to arch.

- Related ticket: N/A
- Needles: N/A
- Verification runs:
  - x86_64: https://openqa.suse.de/tests/4794424
  - s390x: https://openqa.suse.de/tests/4794425
  - ppc64le: https://openqa.suse.de/tests/4794426

### CODE REVIEW

**Review by dzedro — APPROVED** [review]
I would call it workaround, serial  terminal has to be fixed.

**Comment by nicstange** [comment]
I agree with @dzedro, but from a purely functional POV this change looks good to me -> no objections.

File: tests/kernel/install_klp_product.pm [hunk 1]
```diff
@@ -19,6 +19,20 @@ use testapi;
...
+
+sub do_reboot {
+    my $self = shift;
+
+    power_action('reboot', textmode => 1);
+    $self->wait_boot;
+
+    if (is_s390x) {
+        select_console('root-console');
+    } else {
```
Comment (pevik) [inline tests/kernel/install_klp_product.pm:32]: Maybe here just for sure add `prepare_serial_console()` to fix ppc64le issues on older releases?

**Review by pevik — APPROVED** [review]

**Comment by mdoucha** [comment]
`prepare_serial_console()` added, new verification runs:
- x86_64: https://openqa.suse.de/tests/4794797
- s390x: https://openqa.suse.de/tests/4794798
- ppc64le: https://openqa.suse.de/tests/4794799

Note: the kernel maintenance incident was already released so all the verification runs are expected to fail. The important thing is that the s390 verification does not fail with this specific error: https://openqa.suse.de/tests/4789514#step/install_klp_product/32

**Comment by pevik** [comment]
Yes, https://openqa.suse.de/tests/4789514#step/install_klp_product/32 will be fixed with this PR.
@dzedro please, merge this.

**Pull request merged**